ntoskrnl/se/sid.c

```diff
@@ -637,6 +721,7 @@ SeCaptureSidAndAttributesArray(
         }
     }
 
+Cleanup:
     /* Check for failure */
     if (!NT_SUCCESS(Status))
     {
```

Comment:
The line below `if (SidAndAttributes != AllocatedMem)` needs to check for NULL, which can be the case in the `STATUS_BUFFER_TOO_SMALL` code path.